src/app/dashboard/accounts/page.tsx

```diff
@@ -1,21 +1,44 @@
 'use client';
 
 import React from 'react';
-import { mutate } from 'swr';
 import useSWRImmutable from 'swr/immutable';
-import { useDataSource } from '@/hooks';
+import { DateTime, Interval } from 'luxon';
 
+import type { Account } from '@/book/entities';
+import Money from '@/book/Money';
 import AccountsTable from '@/components/AccountsTable';
 import AddAccountButton from '@/components/buttons/AddAccountButton';
-import { getAccountsWithPath } from '@/book/queries';
+import { getAccountsWithPath, getEarliestDate } from '@/book/queries';
 import { PriceDB, PriceDBMap } from '@/book/prices';
+import DateRangeInput from '@/components/DateRangeInput';
+import type { AccountsTree } from '@/types/accounts';
 
 export default function AccountsPage(): JSX.Element {
-  const { save } = useDataSource();
+  const { data: earliestDate } = useSWRImmutable(
+    '/api/start-date',
+    () => getEarliestDate(),
+  );
+
+  React.useEffect(() => {
+    if (earliestDate) {
+      setDateRange(Interval.fromDateTimes(
+        earliestDate,
+        DateTime.now(),
+      ));
+    }
+  }, [earliestDate]);
+
+  const [dateRange, setDateRange] = React.useState(
+    Interval.fromDateTimes(
+      earliestDate || DateTime.now().startOf('year'),
+      DateTime.now(),
+    ),
+  );
+
   let { data: accounts } = useSWRImmutable(
     '/api/accounts/splits',
     () => getAccountsWithPath({
-      relations: { splits: true },
+      relations: { splits: { fk_transaction: true } },
```

Comment:
The data is needed so we can filter transactions/splits when the date ranges change.